bionic/protocols.py

```diff
@@ -877,7 +877,11 @@ def validate(self, value):
         try:
             items = tuple(value)
         except TypeError as e:
-            raise AssertionError(str(e))
+            message = f"""
+            Expected a sequence with length {self._expected_length};
+            did you mean to use @output instead of @outputs?
```

Comment:
Like below, let's also include some description of what we got instead, like `instead, got non-sequence value {value!r}`, to make it easier to debug.

Oh, also, I should have clarified this before, but the ` did you mean to use @output instead of @outputs?` message only makes sense if `self._expected_length` is `1`, right?

Reply:
Ah yes, that was silly of me not to notice! I've amended my commit to fix this.